### PR TITLE
Add Connect Device in LAVA [REVPI-2653]

### PR DESCRIPTION
**Pull request by RamiGspo**
In order to test Connect devices with LAVA, "lava-test-case" has been updated.

### CODE REVIEW

File: automated/linux/ethernet/test-eth-ethtool.yaml [hunk 1]
```diff
@@ -13,7 +13,13 @@ metadata:
         - Core 3
         - core 3+
         - Core S
+        - Connect
+        - Connect+
+        - Connect S
+        - Connect SE
+params:
```
Comment (ferdrich-kunbus) [inline automated/linux/ethernet/test-eth-ethtool.yaml]: How does this work with other devices? What happens, if a Core is tested? Will the test run or fail? How are the devices differentiated and when?

File: automated/linux/leds/test-leds1.yaml [hunk 2]
```diff
@@ -13,7 +13,13 @@ metadata:
         - Core 3
         - core 3+
         - Core S
+        - Connect
+        - Connect+
+        - Connect S
+        - Connect SE
+params:
```
Comment (ferdrich-kunbus) [inline automated/linux/leds/test-leds1.yaml]: Same as above, what happens, when we want to test a Core with this changes testcase?

File: automated/linux/leds/test-leds1.sh [hunk 3]
```diff
@@ -11,7 +11,26 @@ declare -a LEDS_CORE=(
 	"/sys/class/leds/power_red"
 )
 
-for i in "${LEDS_CORE[@]}"
+declare -a LEDS_CONNECT=(
```
Comment (ferdrich-kunbus) [inline automated/linux/leds/test-leds1.sh:19]: Pull that array out into a led_base.sh or something like that, as the same array is used in the other LED test. The path /sys/class/leds/ can be added in the script like it is done in test-leds2.sh.

Maybe we will combine the LED tests into one test later on. Writing to a file for an LED that does not exist will fail like a test if for the existence of a path/file. Could also be tested beforehand.

File: automated/linux/leds/test-leds2.sh [hunk 4]
```diff
@@ -4,19 +4,34 @@ TEST_CASE_NAME=$(basename "$0" .sh)
...
+declare -a LEDS_CONNECT=(
+	"a1_green"
+	"a1_red"
+	"a2_green"
+	"a2_red"
+	"a3_green"
+	"a3_red"
+)
+
+if [ "$1" = "Connect" ]
```
Comment (ferdrich-kunbus) [inline automated/linux/leds/test-leds2.sh]: A switch-case would scale better for upcoming devices and also has a default, if the devicename is missspelled, for example.

File: automated/linux/leds/test-leds1.sh [hunk 5]
```diff
@@ -11,7 +11,26 @@ declare -a LEDS_CORE=(
...
+	"/sys/class/leds/a2_green"
+	"/sys/class/leds/a2_red"
+	"/sys/class/leds/a3_green"
+	"/sys/class/leds/a3_red"
+	"/sys/class/leds/default-on"
+	"/sys/class/leds/mmc0"
+	"/sys/class/leds/power_red"
+)
+
+if [ "$1" = "Connect" ]
```
Comment (ferdrich-kunbus) [inline automated/linux/leds/test-leds1.sh]: Better use a switch-case, as mentioned in the other LED test.

File: automated/device-config/connect/config_loader.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+TEST_CASE_NAME=$(basename "$0" .sh)
+
+if ! cp _config.rsc /var/www/revpi/pictory/projects/_config.rsc;
```
Comment (ferdrich-kunbus) [inline automated/device-config/connect/config_loader.sh]: Semicolon is not needed for the if, if then is in the next line. :D 

Either do 
"if x; then ..."
or
"if x
then ..."

And do it the same way in all scripts.

File: automated/linux/leds/test-leds2.yaml [hunk 7]
```diff
@@ -13,7 +13,14 @@ metadata:
         - Core 3
         - core 3+
         - Core S
+        - Connect
+        - Connect+
+        - Connect S
+        - Connect SE
+
+params:
```
Comment (ferdrich-kunbus) [inline automated/linux/leds/test-leds2.yaml]: Even after the third or fourth time I don't understand, how other devices may be tested with this testcase where the test gets "connect" as parameter, even if a Core is tested. Please explain it to a dumb colleague. ;)

File: automated/device-config/connect/config_loader.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    scope:
+        - functional
+    devices:
+        - Connect
+        - Connect+
+        - Connect S
+        - Connect SE
+run:
+    steps:
+        - cd automated/device-config/connect
```
Comment (ferdrich-kunbus) [inline automated/device-config/connect/config_loader.yaml:18]: Is there a reason or a necessity to first switch to the folder and then execute the test? Would "bash automated/device-config/connect/config_loader.sh" not also work? Please give an explanation.

File: automated/linux/ethernet/test-eth-ethtool.yaml (hunk 1 above)
Comment (RamiGspo) [reply automated/linux/ethernet/test-eth-ethtool.yaml]: That is just information... Metadata is not used by LAVA. It is just for us.
Our devices have "tags". In order to run a test in one device we use this "tags" in the job.

I added some tags in the testsuite as well. For this reason there are two testsuites: for connect and core.
For Connect devices the Testsuite is: connect-testsuite-001.yml
Hier you find this: 
`tags:
- config_connect_01`

For Core devices the Testsuite is: core-testsuite-001.yml
Here we use this tag:
`tags:
- config_core_01`

With these tags, jobs will only run on devices with the same tags.

We defined more tags, but I only use this one for tests and development to identify another devices or another developer.
The tags "config_connect_01" and "config_core_01" should be used by the devices in the rack.

If a device is not available, the job is scheduled.

File: automated/linux/leds/test-leds1.yaml (hunk 2 above)
Comment (RamiGspo) [reply automated/linux/leds/test-leds1.yaml]: See here please: [https://github.com/RevolutionPi/LAVA-test-definitions/pull/14#discussion_r1001488371](url)

File: automated/linux/leds/test-leds2.yaml (hunk 7 above)
Comment (RamiGspo) [reply automated/linux/leds/test-leds2.yaml]: See here please: [https://github.com/RevolutionPi/LAVA-test-definitions/pull/14#discussion_r1001488371](https://github.com/RevolutionPi/LAVA-test-definitions/pull/url)

File: automated/device-config/connect/config_loader.yaml (hunk 8 above)
Comment (RamiGspo) [reply automated/device-config/connect/config_loader.yaml:18]: I think it is the same.
I got this format from the Linaro repository and they used this format in their tests. 
Here are some examples:
[https://github.com/Linaro/test-definitions/blob/master/automated/linux/crypto/crypto.yaml](https://github.com/Linaro/test-definitions/blob/master/automated/linux/crypto/crypto.yaml)
[https://github.com/Linaro/test-definitions/blob/master/automated/linux/cyclictest/cyclictest.yaml](https://github.com/Linaro/test-definitions/blob/master/automated/linux/cyclictest/cyclictest.yaml)
[https://github.com/Linaro/test-definitions/blob/master/automated/linux/ethernet/ethernet.yaml](https://github.com/Linaro/test-definitions/blob/master/automated/linux/ethernet/ethernet.yaml)

Should I used your format? 
I have not tested this, but if it works, it is not a problem to change it.

File: automated/linux/leds/test-leds1.sh (hunk 3 above)
Comment (RamiGspo) [reply automated/linux/leds/test-leds1.sh:19]: I could combine the LED tests now in one script and use a variable with the pfad /sys/class/leds/...
The lava-test-cases are better identified...

File: automated/linux/ethernet/test-eth-ethtool.yaml (hunk 1 above)
Comment (ferdrich-kunbus) [reply automated/linux/ethernet/test-eth-ethtool.yaml]: I don't mean the meta data, I mean the params. The yaml defines a param "DUT" with content "connect", which is then used as parameter for the test script itself. I don't understand how this parameter can be changed during runtime to get also the possibility to test other devices.

At the moment it looks to me that every testrun that will use this test will run it as device "Connect", even if the tests are executed on a Core or a Flat.

File: automated/device-config/connect/config_loader.yaml (hunk 8 above)
Comment (ferdrich-kunbus) [reply automated/device-config/connect/config_loader.yaml:18]: Links are not working...

If Linaro does it this way, then we should adapt it. I was only curious why a cd into the directory was made. One explanation would be, that if there are files and other stuff in the subdirectory to which the script needs to have access to, it is easier to executed it from that dir.

File: automated/linux/ethernet/test-eth-ethtool.yaml (hunk 1 above)
Comment (RamiGspo) [reply automated/linux/ethernet/test-eth-ethtool.yaml]: Oh Yes.. you are right...
In this case more files will be needed...? One for core, one for connect, one for flat... etc...?
And then they will be used in the testsuite. Each device have a testsuite...

Or with Parameters in the Job yaml? They will be used in the Test yaml... and then in the script...
But I did not see Parameters in Yaml from Jobs to Tests...

Maybe more than one file is the better idea. What do you think?

Comment (ferdrich-kunbus) [reply automated/linux/ethernet/test-eth-ethtool.yaml]: It is possible to add parameters in the yaml test definition: https://master.lavasoftware.org/static/docs/v2/actions-test.html#params

Can you check if that would work? A job is dedicated to one device, so it is the correct place to add this parameter.

I don't have strong feelings at the moment for one of the solutions. Both parameter based and splitted tests have some pros and cons. I would try it with the parameter. If we see some problems with it, we can change it to another solution.

File: automated/device-config/connect/config_loader.yaml (hunk 8 above)
Comment (RamiGspo) [reply automated/device-config/connect/config_loader.yaml:18]: I updated the Links... sorry.
It should work...

File: automated/linux/ethernet/test-eth-ethtool.yaml (hunk 1 above)
Comment (RamiGspo) [reply automated/linux/ethernet/test-eth-ethtool.yaml]: OK... it works...
I will use it. The parameters are now in the Job and not in the Test.

File: automated/linux/leds/test-leds1.sh [hunk 9]
```diff
@@ -1,22 +1,66 @@
...
+
+	if echo 1 > "$i"/brightness
+	then
+		lava-test-case "$TEST_CASE_NAME-$i/brightness-green" --result pass
+	else
+		lava-test-case "$TEST_CASE_NAME-$i/brightness-green" --result fail
+	fi
+	
+	sleep $LED_TIME
+		if echo 0 > "$i"/brightness
```
Comment (ferdrich-kunbus) [inline automated/linux/leds/test-leds1.sh]: Indentation error

Comment (RamiGspo) [reply automated/linux/leds/test-leds1.sh]: Thank you!

**Review by ferdrich-kunbus — APPROVED** [review]
mergeable

**Pull request merged**